src/kixi/nhs/data/transform.clj

```diff
@@ -107,6 +107,22 @@
   [data recipe]
   (update-in data [:value] process-percentages))
 
+;; Calculate confidence intervals:
+;; when several time periods in
+;; the same data resource in ckan
+(defn calculate-lci-uci
+  [condition data]
+  (if condition
+    (let [sorted-data (reverse (sort-by :end_date data))
+          current (Float/parseFloat (:value (first sorted-data)))
+          previous (Float/parseFloat (:value (second sorted-data)))
```

Comment:
It would be good to add a check here that doesn't try to parse null values, as they will cause `NullPointerException`.